examples/persistent-storage/qpg6100/BUILD.gn

```diff
@@ -0,0 +1,77 @@
+# Copyright (c) 2021 Project CHIP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import("//build_overrides/build.gni")
+import("//build_overrides/chip.gni")
+import("//build_overrides/qpg6100_sdk.gni")
+
+import("${build_root}/config/defaults.gni")
+
+import("${qpg6100_sdk_build_root}/qpg6100_executable.gni")
+import("${qpg6100_sdk_build_root}/qpg6100_sdk.gni")
+
+assert(current_os == "freertos")
+
+qpg6100_project_dir = "${chip_root}/examples/persistent-storage/qpg6100"
+examples_plat_dir = "${chip_root}/examples/platform/qpg6100"
+
+qpg6100_sdk("sdk") {
+  include_dirs = [
+    "${chip_root}/src/platform/qpg6100",
+    "${examples_plat_dir}/project_include",
+  ]
+
+  sources = [ "${qpg6100_project_dir}/include/CHIPProjectConfig.h" ]
+
+  defines = []
+}
+
+qpg6100_executable("persistent_storage") {
+  include_dirs = [ "${qpg6100_project_dir}/.." ]
+  defines = []
+  output_name = "chip-qpg6100-persistent_storage-example.out"
+
+  deps = []
+
+  public_deps = [
+    ":sdk",
+    "${chip_root}/src/lib",
+  ]
+
+  include_dirs += [
+    "${qpg6100_project_dir}/include",
+    "${chip_root}/src/app/util",
```

Comment:
Please avoid directly adding other targets' include directories, so remove `"${chip_root}/src/app/util",` here. Use dependencies instead.

Reply:
Thanks for the feedback. I checked other platforms and they do the same. I tried adding this in dependencies, but it doesn't help finding the directory in include paths. Can you give an example where I should add this so it can find it in the include path ?